azimuth_caas_operator/utils/cluster.py

```diff
@@ -4,12 +4,33 @@
 import yaml
 
 from azimuth_caas_operator.models import registry
+from azimuth_caas_operator.models.v1alpha1 import cluster as cluster_crd
 from azimuth_caas_operator.utils import image as image_utils
 
 
 LOG = logging.getLogger(__name__)
 
 
+async def ensure_cluster_id(client, cluster: cluster_crd.Cluster):
+    """
+    Ensures that the given cluster has it's ID set.
+    """
+    if cluster.status.clusterID:
+        return
+    # Only update the status of the cluster object we were given if the
+    # patch is successful
+    name = cluster.metadata.name
+    namespace = cluster.metadata.namespace
+    cluster_resource = await client.api(registry.API_VERSION).resource("cluster")
+    await cluster_resource.patch(
+        name,
+        {"status": {"clusterID": cluster.metadata.uid}},
+        namespace=namespace,
+    )
+    cluster.status.clusterID = cluster.metadata.uid
```

Comment:
Hmm, this is the line I meant to post my comment against. It seems odd mutating the object passed in.

Reply:
I don't mind it. The object is unique to the invocation of the handler anyway, not global, so 🤷